File: back2/api/v1/services/jwt_service.py
import os, time, jwt

class JWTService:
    def __init__(self):
        self.secret = os.getenv("JWT_SECRET", "dev-secret-change-me")
        self.exp_seconds = int(os.getenv("JWT_EXP_SECONDS", "3600"))

    def get_expiration_seconds(self) -> int:
        return self.exp_seconds

    def generate_token(self, usuario) -> str:
        now = int(time.time())
        payload = {
            "sub": str(usuario.id),
            "email": usuario.email,
            "rol": int(usuario.rol),   # si Rol es IntEnum
            "iat": now,
            "exp": now + self.exp_seconds
        }
        return jwt.encode(payload, self.secret, algorithm="HS256")

File: back/src/main/java/com/example/ss1/services/JWTService.java
package com.example.ss1.services;

import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import io.jsonwebtoken.security.Keys;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import com.example.ss1.models.Usuario;

import java.nio.charset.StandardCharsets;
import java.security.Key;
import java.util.Date;

@Service
public class JWTService {
      @Value("${jwt.secret}")
    private String secret;

    @Value("${jwt.expiration}")
    private long expirationMs;

    private Key getSigningKey() {
        // puedes usar secret.getBytes(StandardCharsets.UTF_8) si la clave es suficientemente larga
        return Keys.hmacShaKeyFor(secret.getBytes(StandardCharsets.UTF_8));
    }

    public String generateToken(Usuario usuario) {
        Date now = new Date();
        Date expiryDate = new Date(now.getTime() + expirationMs);

        return Jwts.builder()
                .setSubject(usuario.getId().toString()) // id del usuario
                .claim("email", usuario.getEmail())
                .claim("rol", usuario.getRol().name())   // asumiendo Rol es un enum
                .setIssuedAt(now)
                .setExpiration(expiryDate)
                .signWith(getSigningKey(), SignatureAlgorithm.HS256)
                .compact();
    }

    public long getExpirationSeconds() {
        return expirationMs / 1000;
    }
}
